pkg/ruler/api.go

```diff
@@ -496,6 +496,16 @@ func (a *API) CreateRuleGroup(w http.ResponseWriter, req *http.Request) {
 	}
 
 	rgProto := rulespb.ToProto(userID, namespace, rg)
+	loadedRg := rulespb.FromProto(rgProto)
+	rgYaml, err := yaml.Marshal(loadedRg)
+	if err == nil {
+		err = yaml.Unmarshal(rgYaml, &rulefmt.RuleGroup{})
```

Comment:
In which case there is no error for `yaml.Marshal` but we got error from `yaml.Unmarshal`?
Can you add this into the unit test?